src/Components/Aspire.RabbitMQ.Client/AspireRabbitMQExtensions.cs

```diff
@@ -149,16 +168,12 @@ public Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, Canc
         }
     }
 
-    private static IConnection CreateConnection(IConnectionFactory factory, int retryCount)
+    private static IConnection CreateConnection(IConnectionFactory factory, ResiliencePipeline resiliencePipeline)
     {
-        var policy = Policy
-            .Handle<SocketException>().Or<BrokerUnreachableException>()
-            .WaitAndRetry(retryCount, retryAttempt => TimeSpan.FromSeconds(Math.Pow(2, retryAttempt)));
-
         using var activity = s_activitySource.StartActivity("rabbitmq connect", ActivityKind.Client);
         AddRabbitMQTags(activity);
 
-        return policy.Execute(() =>
+        return resiliencePipeline.Execute(() =>
```

Comment:
If you pass the `factory` value through here as state to the `Execute()` call, you can make the delegate `static` and avoid the need to allocate the closure.

https://www.pollydocs.org/advanced/performance.html#use-static-lambdas